.ci/docker/golang-mage/Dockerfile

```diff
@@ -9,14 +9,14 @@ RUN git clone https://github.com/magefile/mage \
   && go run bootstrap.go \
   && mage -version \
   && cd /tools \
-  && go get github.com/elastic/apm-server/vendor/golang.org/x/tools/cmd/benchcmp \
-  && go get github.com/elastic/apm-server/vendor/github.com/t-yuki/gocover-cobertura \
-  && go get github.com/elastic/apm-server/vendor/github.com/jstemmer/go-junit-report \
-  && go get github.com/elastic/apm-server/vendor/github.com/kardianos/govendor \
+  && go get golang.org/x/tools/cmd/benchcmp \
```

Comment:
These commands aren't likely to be effective, as we no longer use the tools installed in $GOBIN, but rather we pin to specific module versions and install them into the build directory. I think we should use the approach of running `go mod download`, like I added to `tests/Dockerfile`.

To do that we would need to make these additions to the Dockerfile:

- COPY go.mod /tools
- RUN go mod download

And remove all of this entire RUN command, i.e. git clone, mage, go get.

We would then need to update Jenkinsfile, passing in the repo root as the build context:

```diff
diff --git a/Jenkinsfile b/Jenkinsfile
index 7fa2c466..8a666436 100644
--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -419,7 +419,7 @@ def golang(Closure body){
   def golangDocker
   retry(3) { // Retry in case there are any errors when building the docker images (to avoid temporary glitches)
     sleep randomNumber(min: 2, max: 5)
-    golangDocker = docker.build('golang-mage', "--build-arg GO_VERSION=${GO_VERSION} ${BASE_DIR}/.ci/docker/golang-mage")
+    golangDocker = docker.build('golang-mage', "--build-arg GO_VERSION=${GO_VERSION} -f .ci/docker/golang-mage/Dockerfile ${BASE_DIR}")
   }
   withEnv(["HOME=${WORKSPACE}", "GOPATH=${WORKSPACE}", "SHELL=/bin/bash"]) {
      golangDocker.inside('-v /usr/bin/docker:/usr/bin/docker -v /var/run/docker.sock:/var/run/docker.sock'){
```

Reply:
Having said that, that's just what I think is ideal. Feel free to merge this PR, as it will unblock master.

Reply:
Thanks for all the details. I will merge this in for now, so master is green again and then come up with a follow-up PR.